### PR TITLE
Fix CI pipeline

### PR DESCRIPTION
**Pull request by SimonBaeumer**
Fixes #

## Checklist

 - [ ] Added unit / integration tests for windows, macOS and Linux? 
 - [ ] Added a changelog entry in [CHANGELOG.md](https://github.com/commander-cli/commander/blob/master/CHANGELOG.md)?
 - [ ] Updated the documentation ([README.md](https://github.com/commander-cli/commander/blob/master/README.md), [docs](https://github.com/commander-cli/commander/blob/master/docs))?
 - [ ] Does your change work on `Linux`, `Windows` and `macOS`?

### CODE REVIEW

File: commander_unix.yaml [hunk 1]
```diff
@@ -126,6 +126,7 @@ tests:
     exit-code: 0
 
   it should execute test from url:
+    skip: true #TODO: circle ci does not resolve the URL properly
```
Comment (dylanhitt) [inline commander_unix.yaml:129]: I may not have found the right pipeline failure (I’m only on a phone at the moment), but I can only find where the name resolution for the image pull is failing. Did you mean to remove this?

**Comment by dylanhitt** [comment]
I got around to looking deeper into the DNS problems here. 

I've reduced it down to not attaching the container that `commander` spins up to the network that we create in order to run our integration test. 

We perform the following to set up an env to both test our docker executor and ssh executor. 

```bash
docker network create --subnet=172.28.0.0/16 commander_test

docker run -d \
  --rm \
  --ip=172.28.0.2 \
  --network commander_test \
  --name commander-int-ssh-server \
  commander-int-ssh-server

docker run \
  --rm \
  -v /var/run/docker.sock:/var/run/docker.sock \
  --network commander_test \
  --name commander-integration-go-test \
  --env CC_TEST_REPORTER_ID="${CC_TEST_REPORTER_ID}" \
  --dns=8.8.8.8 \
  commander-int-test \
  make "$MAKE_TARGET"
```

This works for the ssh execution but not docker. If you remove the network settings you will find that the docker execution will work.. [pipeline for example](https://app.circleci.com/pipelines/github/commander-cli/commander/143/workflows/6b6e4810-1cc9-43b0-9c11-8909c98bd074/jobs/507).

So possible solutions.

1. Split the ssh and docker execution tests and run them as different jobs. 
2. Provide an option in commander to setup the docker executor with a pre existing network config. Essentially [this](https://github.com/commander-cli/commander/pull/184/files#diff-f9fd4fb3746ce213c678dc4e7b29b5769ed70af7c05cff7add2a616a4e3d5feaR84)

**Comment by dylanhitt** [comment]
After a lot of head banging. I've come to the conclusion this DNS issues has to be on the circleCI side...

Now we have the problem of these pipelines failing silently but that's for another time. However, here is example of a pipeline working with he [docker properly pulling and running our test](https://app.circleci.com/pipelines/github/commander-cli/commander/172/workflows/608fb526-845a-4fb3-9edb-335443923540/jobs/593). I purposely changed the image over to the golang image that we use in the docker builds. You can see that the image was pulled/ran. I then made [this minor change](https://github.com/commander-cli/commander/commit/a96ea871679b7379681154dbf46a0744788f2edb) to get the pipeline to pass. The very next pipeline fails with [DNS name resolution problems](https://app.circleci.com/pipelines/github/commander-cli/commander/173/workflows/a44467f1-1189-457a-b217-ff3023b7d389/jobs/597).

You can look through all the pipelines from the `bump-docker-version` and find sporadic successes. I'm a bit lost. Maybe in a few weeks it will resolve 🤷

**Comment by dylanhitt** [comment]
One solution could be separating out the ssh test and docker tests. SSH seems to be fine within the containers and all docker commands seem to be more stable outside of the container... 

However I was picking up random docker pull failures for the alpine image even outside the dockerized thes.. [see this](https://app.circleci.com/pipelines/github/commander-cli/commander/164/workflows/1dfaa06e-f72d-49e9-8563-5689df8cfd45/jobs/569), here is the line it failed [on](https://github.com/commander-cli/commander/commit/c9fed9ac0d94b6f0d9ef46b81dd3a2c6dd264).

It seems though running as separate jobs is fine. I don't know. This seems a bit strange to me 🤔

File: commander_unix.yaml (hunk 1 above)
Comment (SimonBaeumer) [reply commander_unix.yaml:129]: I saw this at least sometime to be flaky because of DNS issues, can't show you a build though I think :/

Comment (dylanhitt) [reply commander_unix.yaml:129]: Yeah, I saw it in my testing too..

**Comment by dylanhitt** [comment]
@SimonBaeumer do you have any opinions on proceeding? My only solution would be to separate out the docker execution test from the ssh test. It seems like circle ci really hates the docker executor spinning up another container with in a container. 

To be brief, running the docker integration as it’s own job seems to work all the time.

**Comment by dylanhitt** [comment]
Hey @SimonBaeumer CircelCI seems to be behaving now. It seems the latest [bump-docker](https://app.circleci.com/pipelines/github/commander-cli/commander?branch=bump-docker-version&filter=all) pipelines are passing.

You want to give this. another shot.

**Review by dylanhitt — APPROVED** [review]
Fixed the script to get it running. Mainly just putting the COPY back in the dockerfile even though mounting is the better option.. Not sure why circleci doesn't like that.

**Pull request merged**